File: src/main/FSharp/day1/day1.fs
// MIT License

// Copyright (c) 2022 Ian Russell

// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:

// The above copyright notice and this permission notice shall be included in all
// copies or substantial portions of the Software.

// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

open System.IO

let lines = File.ReadAllText "./day1/input.txt"


let calories = lines.Split("\r\n\r\n")
                    |> Array.map(fun x -> x.Split("\r\n")|> Array.sumBy int)

// Part 1
let maxCalories = calories |> Array.max


// Part 2
let sumofTopThree = calories |> Array.sortByDescending id |> Array.take 3 |> Array.sum

File: Python/aoc2021/day1/day1.py
# Part 1

input_file = open("input.txt", "r")
data = input_file.read().split("\n")
input_file.close()

counter = 0

for i in range(1, len(data)):
    if int(data[i]) > int(data[i - 1]):
        counter = counter + 1

print("Answer for part 1: %2d" % counter)


# Part 2

def count_three_measurement_window(index):
    return sum(int(data[index]) for index in (index, index + 2))


counter = 0
for i in range(0, len(data) - 2):
    if count_three_measurement_window(i) > count_three_measurement_window(i - 1):
        counter = counter + 1

print("Answer for part 2: %2d" % counter)
